src/store/Entity/EntityStripper.ts

```diff
@@ -0,0 +1,101 @@
+import Dictionary from '@/common/interfaces/Dictionary';
+import InvalidEntityException from '@/store/Entity/exceptions/InvalidEntityException';
+
+export default class EntityStripper {
+	private unstrippedEntity: any;
+
+	constructor( entity: any ) {
+		if ( typeof entity === 'string' ) {
+			entity = JSON.parse( entity );
+		}
+
+		this.unstrippedEntity = entity;
+		this.checkPrerequisites();
+	}
+
+	public getId(): string {
+		return this.unstrippedEntity.id;
+	}
+
+	public getType(): string {
+		return this.unstrippedEntity.type;
+	}
+
+	public getLabels(): Dictionary<string> {
+		return this.getNestedLabels();
+	}
+
+	public getDescriptions(): Dictionary<string> {
+		return this.getNesatedDescriptions();
+	}
+
+	public getAliases(): Dictionary<string[]> {
+		return this.getNestedAliases( this.unstrippedEntity );
+	}
+
+	private getNestedValues( entityObject: any, index: string ): Dictionary<string> {
+		const value: Dictionary<string> = {};
+
+		if ( !entityObject.hasOwnProperty( index ) ) {
+			return value;
+		}
+		Object.keys( entityObject[ index ] ).forEach( ( languageKey: string ) => {
+			value[ languageKey ] = entityObject[ index ][ languageKey ].value;
+		} );
+
+		return value;
+	}
+
+	private getNestedLabels(): Dictionary<string> {
+		return this.getNestedValues( this.unstrippedEntity, 'labels' );
+	}
+
+	private getNesatedDescriptions(): Dictionary<string> {
+		return this.getNestedValues( this.unstrippedEntity, 'descriptions' );
+	}
+
+	private checkPrerequisites(): void {
```

Comment:
IMO this is a bit overly cautious and could get away with checking that `type` and `id` exist on the given JSON object. The rest should ideally be taken care of by strict typing.